### ISSUE TITLE
onGetErrorMessage in <TextField> displays error twice.

### ISSUE THREAD

**Issue by carruthe** [OfficeDev/office-ui-fabric-react#1240]
![doubleerror](https://cloud.githubusercontent.com/assets/23384422/23797306/39fca8a2-056d-11e7-9fc9-8449e190c041.jpeg)

### Bug Report

- __Package version(s)__: 2.1.0
- __Browser and OS versions__: MacOS Sierra, Safari, Firefox, Chrome

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No) Not up to speed on pull requests yet

Requested priority: (Blocking, High, Normal, Low) High

#### Describe the issue:

Returning an error message from `onGetErrorMessage` in <TextField> component results in the error message being displayed twice, once formatted and once not.

### If applicable, please provide a codepen repro:

https://codepen.io/tcarruthers/pen/ryyvpq

Type in more than three letters in the field to activate the error message.

**Comment by dzearing** [comment]
This isn't totally a bug but it isn't great either; you need fabric-core css. The extra div being rendered is a screenreader only text element. The `ms-u-screenreaderonly` class is not being defined so the element isn't being hidden.

I've created a fork of your codepen and included the styles
https://codepen.io/dzearing/pen/dvvqao?editors=1010

**Comment by dzearing** [comment]
@micahgodbolt @mikewheaton @Jahnp To be honest I don't understand the purpose of the extra screenreaderonly element; shouldn't screen readers catch the actual element containing the actual text?

**Comment by carruthe** [comment]
@dzearing Thanks - I was importing the `office-ui-fabric-core` version of fabric.min.css.  Switching to the one included in `office-ui-fabric-react` fixed it.

Nick

**Comment by micahgodbolt** [comment]
@dzearing I agree that it'd be nice to remove the dependency on core. 

The TextField does this with error messages

```
<div 
  aria-live='assertive' 
  className='ms-u-screenReaderOnly' 
  data-automation-id='error-message'>
    { errorMessage }
</div>

```

> Dynamic content which updates without a page reload is generally either a region or a widget. Simple content changes which are not interactive should be marked as live regions. Below is a list of each related ARIA live region property with a description.
> - aria-live: The aria-live=POLITENESS_SETTING is used to set the priority with which screen reader should treat updates to live regions - the possible settings are: off/polite/assertive. The default setting is 'off'. This attribute is by far the most important.


This all seems valuable. So is there a reason that aria-live can't go on the standard error text? Both divs print the exact same thing `{ errorMessage }`


```
<p
  aria-live='assertive'
  className={ css('ms-TextField-errorMessage ms-u-slideDownIn20', styles.errorMessage) }
>
  { errorMessage }
</p>
```

**Comment by dzearing** [comment]
@micahgodbolt Good point! Not sure why we'd need 2 elements instead of just adding the attribute on there. Maybe the screenreaders don't like dynamically modifying the element or something...

@cschlechty Any thoughts? Can we just condense the message to 1 element without having to have a duplicate hidden "screenreader only" element?

**Comment by cschlechty** [comment]
It was likely done this way to work around a bug where adding the live
region and message all at once won't get read as it's not considered an
update to the region.

**Comment by micahgodbolt** [comment]
@cschlechty is that bug still an issue? Should we maintain this as a feature of the textfield?

**Comment by mdahamiwal** [comment]
We can workaround that with performing an update to live region. We fixed that for MessageBar with delay rendering the content in a live region with a small delay of 10ms. The screenreader will understand that as an update. 

+1 for removing the redundant screen reader specific div and use the actual error div as live region. 
I can float a PR for these changes if required.

**Comment by mikewheaton** [comment]
If I'm understanding this right, the bug was fixed in #1287?
